httpserver/src/main/kotlin/tbdex/sdk/httpserver/handlers/GetBalances.kt

```diff
@@ -0,0 +1,105 @@
+package tbdex.sdk.httpserver.handlers
+
+import io.ktor.http.HttpHeaders
+import io.ktor.http.HttpStatusCode
+import io.ktor.server.application.ApplicationCall
+import io.ktor.server.response.respond
+import tbdex.sdk.httpclient.RequestToken
+import tbdex.sdk.httpclient.models.ErrorDetail
+import tbdex.sdk.httpserver.models.BalancesApi
+import tbdex.sdk.httpserver.models.CallbackError
+import tbdex.sdk.httpserver.models.ErrorResponse
+import tbdex.sdk.httpserver.models.GetBalancesCallback
+import tbdex.sdk.protocol.models.Balance
+
+/**
+ * Get balances response
+ *
+ * @property data list of Balances
+ */
+class GetBalancesResponse(
+  val data: List<Balance>?
+)
+/**
+ * Get balances request handler
+ *
+ * @param call Ktor server application call
+ * @param balancesApi Balances API interface
+ * @param callback Callback function to be invoked
+ */
+@Suppress("SwallowedException")
+suspend fun getBalances(
+  call: ApplicationCall,
+  balancesApi: BalancesApi,
+  callback: GetBalancesCallback?,
+  pfiDid: String
+) {
+
+  val authzHeader = call.request.headers[HttpHeaders.Authorization]
+  if (authzHeader == null) {
+    call.respond(
+      HttpStatusCode.Unauthorized,
+      ErrorResponse(
+        errors = listOf(
+          ErrorDetail(
+            detail = "Authorization header required"
+          )
+        )
+      )
+    )
+    return
+  }
+
+  val arr = authzHeader.split("Bearer ")
+  if (arr.size != 2) {
+    call.respond(
+      HttpStatusCode.Unauthorized,
+      ErrorResponse(
+        errors = listOf(
+          ErrorDetail(
+            detail = "Malformed Authorization header. Expected: Bearer TOKEN_HERE"
+          )
+        )
+      )
+    )
+    return
+  }
+
+  val token = arr[1]
+  val requesterDid: String
```

Comment:
I haven't used `ktor` before but if it has the concept of middleware, maybe it'd be helpful to abstract this logic out into a `verifyAuthzHeader` middleware? and if `ktor` doesn't have the middleware concept, could just tuck this into a method and use it for all protected endpoints. (note for future PR)